frontend/src/models/propertyDefinitionsModel.ts

```diff
@@ -107,18 +108,32 @@ const checkOrLoadPropertyDefinition = (
     propertyName: BreakdownKeyType | undefined,
     definitionType: PropertyDefinitionType,
     propertyDefinitionStorage: PropertyDefinitionStorage,
+    eventMetadataPropertyDefinitions: PropertyDefinition[],
     groupTypeIndex?: number | null
 ): PropertyDefinition | null => {
     // first time we see this, schedule a fetch
     const key = getPropertyKey(definitionType, propertyName, groupTypeIndex)
     if (typeof propertyName === 'string' && !(key in propertyDefinitionStorage)) {
-        window.setTimeout(
-            () =>
-                propertyDefinitionsModel
-                    .findMounted()
-                    ?.actions.loadPropertyDefinitions([propertyName], definitionType, groupTypeIndex),
-            0
-        )
+        if (definitionType === PropertyDefinitionType.EventMetadata) {
+            propertyDefinitionsModel
+                .findMounted()
+                ?.actions.updatePropertyDefinitions(
+                    Object.fromEntries(
+                        eventMetadataPropertyDefinitions.map((definition) => [
+                            `${PropertyDefinitionType.EventMetadata}/${definition.id}`,
+                            definition,
+                        ])
+                    )
+                )
```

Comment:
It's loading all sorts of random stuff now.

![image](https://github.com/user-attachments/assets/8f88c2bc-3d00-4bc2-af04-37d9ba5ff367)